lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -285,13 +297,13 @@ def _group_name(self, root_path: str, file_path: str, group_name: str) -> str:
         #  - name, from juju topology
         #  - suffix, from the relative path of the rule file;
         group_name_parts = [self.topology.identifier] if self.topology else []
-        group_name_parts.extend([rel_path, group_name, "alerts"])
+        group_name_parts.extend([rel_path, group_name, f"{self.rule_type}s"])
         # filter to remove empty strings
         return "_".join(filter(None, group_name_parts))
 
     def _is_already_modified(self, name: str) -> bool:
         """Detect whether a group name has already been modified with juju topology."""
-        modified_matcher = re.compile(r"^.*?_[\da-f]{8}_.*?alerts$")
+        modified_matcher = re.compile(r"^.*?_[\da-f]{8}_.*?(records|alerts)$")
```

Comment:
What if we get an alert group originally called "too_many_records"?

Reply:
What if we get an alert group called `too_many_deadbeef_alerts` right now? Same breakage.